contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,229 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "../data/DataStore.sol";
 import "../role/RoleModule.sol";
 import "../fee/FeeUtils.sol";
+import "../data/Keys.sol";
+import "../v1/IVaultV1.sol";
+import "./IFeedAddress.sol";
 
 // @title FeeHandler
 contract FeeHandler is ReentrancyGuard, RoleModule {
+    using SafeERC20 for IERC20;
+
     DataStore public immutable dataStore;
     EventEmitter public immutable eventEmitter;
+    address public immutable vault;
+    address public immutable gmx;
+    address public immutable wnt;
 
     constructor(
         RoleStore _roleStore,
         DataStore _dataStore,
-        EventEmitter _eventEmitter
+        EventEmitter _eventEmitter,
+        address _vault,
+        address _gmx,
+        address _wnt
     ) RoleModule(_roleStore) {
         dataStore = _dataStore;
         eventEmitter = _eventEmitter;
+        vault = _vault;
+        gmx = _gmx;
+        wnt = _wnt;
     }
 
     // @dev claim fees from the specified markets
     // @param markets the markets to claim fees from
-    // @param tokens the fee tokens to claim
+    // @param feeTokens the fee tokens to claim for (GMX/WNT)
+    // @param swapTokens the provided swap token (GMX/WNT)
     function claimFees(
-        address[] memory markets,
-        address[] memory tokens
+        address[] calldata markets,
+        address[] calldata feeTokens,
+        address[] calldata swapTokens
     ) external nonReentrant onlyFeeKeeper {
-        if (markets.length != tokens.length) {
-            revert Errors.InvalidClaimFeesInput(markets.length, tokens.length);
+        if (markets.length != feeTokens.length) {
+            revert Errors.InvalidClaimFeesInput(markets.length, feeTokens.length);
+        }
+
+        if (markets.length != swapTokens.length) {
+            revert Errors.InvalidClaimFeesInput(markets.length, swapTokens.length);
         }
 
-        address receiver = dataStore.getAddress(Keys.FEE_RECEIVER);
+        address receiver = address(this);
+
+        (uint256 gmxTokenOraclePrice, uint256 gmxTokenOracleDecimals, uint256 gmxTokenDecimals) = _getTokenPricingValues(gmx);
+        (uint256 wntTokenOraclePrice, uint256 wntTokenOracleDecimals, uint256 wntTokenDecimals) = _getTokenPricingValues(wnt);
```

Comment:
do we need to specify "gmx" and "wnt"? i had the impression that the user would just specify the buybackToken, and all the values would be fetched based on the buybackToken, without explicitly stating the token is gmx or wnt, if there is a buyback batchsize and buyback factor defined for that token, that would mean the token can be bought back, is that possible?

Reply:
Updated so that gmx and wnt are only referenced in the _setAvailableFeeAmounts() private function. 

I'm not sure that it can be avoided in this case given that the buyback factor is defined for gmx and for wnt it is inferred. 

Also, in the claimFees() function the buyback token is no longer specified now that buybackFees() is it's own function, thus when computing the updated available fee amounts for gmx and wnt the tokens need to be explicitly referenced.